### ISSUE TITLE
Remaining Sphinx warnings

### ISSUE THREAD

**Issue by melissawm** [napari/napari#4431]
## 📚 Documentation

Currently, we count  Sphinx warnings when building the docs. This is a bit cumbersome, and also stops us from seeing new Warnings that are created when something goes wrong in the docs build, because these new Warnings are lost in this big output.

There are currently three main kinds of warnings left:
- [ ] ImportErrors when building the docs: Because of how autosummary and autodoc are applied together here, some attributes are not fully qualified when the autosummary template is parsed. A proposal to fix this is here: However, I'm not sure if this is the best solution. I am open to any suggestions!
- [ ] `WARNING: duplicate object description`: again, this is because of how autosummary works, if an attribute is inherited and we expect it to have multiple docstrings (for example, as a parameter on a class docstring, or as a property declared later in the same class or a child class) this generates a duplicate description warning. I don't have an ideal solution for this at the moment.
- [ ] Warnings about docstrings from imported members: there are a few warnings about functools.partial or tqdm, which are created when those docstrings are injected into the napari sphinx docs. We should probably add these functions to a blocklist and not actually document them with autosummary, but rather link to their own documentation. Does this sound reasonable?

If there are any other ideas or opinions on how to fix these autosummary warnings, please share.

**Comment by neuromusic** [comment]
> A proposal to fix this is here

I think you missed the link?

**Comment by melissawm** [comment]
Sorry - just edited the description, but [here it is on my fork](https://github.com/melissawm/napari/commit/ca1d0dbbf66f830ca318cb0977f4d3b08f9ac0b3)

**Comment by neuromusic** [comment]
thank you 😄

**Comment by AhmetCanSolak** [comment]
hello @melissawm ! 

I am not familiar with recent state of sphinx warnings but especially for the third kind  of sphinx warnings adding `autosummary_imported_members = False`  to `conf.py` can be helpful.

**Comment by melissawm** [comment]
@AhmetCanSolak that is right, but in that case many of the currently documented objects in napari will no longer be documented - for example, none of the Layers and several other objects are missing if we do that. A workaround would be to add them manually, but I'm not sure if this would be too cumbersome to maintain in the future.

**Comment by AhmetCanSolak** [comment]
I understand  @melissawm .  Could you point me to a specific imported example in documentation source? I like to play around this to see if we can find a way to handle warnings without causing extra maintenance load to our future-selves.
